database/database.go

```diff
@@ -1,127 +1,585 @@
+}
 
-	rows, err := database.Query("SELECT id,name,note,title FROM teachers")
-	defer rows.Close()
+// StoreTeacher stores a new teacher
+// If the ID is not zero, StoreTeacher will try to find the corresponding teacher and overwrite it
+// If the ID is nil a new teacher will be created
+func StoreTeacher(t TeacherT) error {
+	var err error
 
+	// Verify connection to PostgreSQL database
+	err = postgresDatabase.Ping()
 	if err != nil {
-		log.Print("Cannot get teachers: ", err)
-		return nil, err
+		postgresDatabase.Close()
+		return errors.New("At StoreTeacher: " + err.Error())
 	}
 
-	var teachers []Teacher
+	if t.TeacherID == 0 {
+		// add teacher to postgresDatabase
+		err = postgresDatabase.QueryRow(
+			`INSERT INTO teachers (Name, Title, Note) VALUES ($1, $2, $3) RETURNING TeacherID`,
+			t.Name, t.Title, t.Note).Scan(&t.TeacherID)
+		if err != nil {
+			return errors.New("At StoreTeacher: " + err.Error())
+		}
 
-	for rows.Next() {
-		t := Teacher{}
-		rows.Scan(&t.ID, &t.Name, &t.Note, &t.Title)
+		// add teacher to localDatabase
+		addTeacherToLocalDatabase(t)
+	} else {
+		// try to find corresponding entry postgresDatabase and overwrite it
+		var res sql.Result
+		res, err = postgresDatabase.Exec(
+			`UPDATE teachers SET Name=$2, Title=$3, Note=$4 WHERE TeacherID=$1`,
+			t.TeacherID, t.Name, t.Title, t.Note)
+		if err != nil {
+			return errors.New("At StoreTeacher: " + err.Error())
+		}
+		if rowsAffected, _ := res.RowsAffected(); rowsAffected != 0 {
+			return errors.New("At StoreTeacher: Could not find specified entry for overwrite")
+		}
 
-		teachers = append(teachers, t)
+		// try to find corresponding entry in localDatabase and overwrite it
+		err = overwriteTeacherInLocalDatabase(t)
+		if err != nil {
+			return errors.New("At StoreTeacher: " + err.Error())
+		}
 	}
 
-	return teachers, nil
+	log.Print(localDatabase.teacherSlice)
+
+	return nil
+}
+
+// DeleteTeacher deletes the teacher corresponding to the given ID from the database and the teachers slice
+// It will delete all corresponding quotes
+func DeleteTeacher() {
+
+}
+
+/* -------------------------------------------------------------------------- */
+/*                    EXPORTED UNVERIFIED QUOTES FUNCTIONS                    */
+/* -------------------------------------------------------------------------- */
+
+// GetUnverifiedQuotes returns a slice containing all quotes
+func GetUnverifiedQuotes() {
+
+}
+
+// StoreUnverifiedQuote stores an unverified quote
+func StoreUnverifiedQuote() {
+
+}
+
+// DeleteUnverifiedQuote deletes an unverified quote
+func DeleteUnverifiedQuote() {
+
 }
 
-// GetQuotes is a function to get quotes from the database
-func GetQuotes() ([]Quote, error) {
+/* -------------------------------------------------------------------------- */
+/*                              PRIVATE FUNCTIONS                             */
+/* -------------------------------------------------------------------------- */
+
+func loadLocalDatabase() error {
```

Comment:
terminology `localDatabase` is unfortunate - it's only a local cache, which can be loaded/reloaded